py/torch_tensorrt/dynamo/conversion/converter_utils.py

```diff
@@ -686,3 +676,183 @@ def calculate_strides(shape: Sequence[int]) -> Sequence[int]:
     return strides
+
+
+def broadcast(
+    network: TRTNetwork,
+    a: TRTTensor,
+    b: TRTTensor,
+    a_name: str,
+    b_name: str,
+    preset_diff: int = 0,
+) -> Tuple[TRTTensor, TRTTensor]:
+    """
+    Broadcast two TensorRT tensors to the same number of dimensions by
+    prepending 1s to the tensor with less number of dimensions.
+
+    Args:
+        network (TRTNetwork): TensorRT network object.
+        a (TRTTensor): A TensorRT ITensor.
+        b (TRTTensor): A TensorRT ITensor.
+        a_name (str): Name of tensor a.
+        b_name (str): Name of tensor b.
+        preset_diff (int): The difference of number of dimensions after broadcast.
+            A positive number means after broadcast, tensor `a` would have `preset_diff`
+            more dimensions than `b`. This is used in matmul, since we need to broadcast
+            tensors but not always to the same number of dimension. The reason is that
+            matmul supports Matrix x Vector and in this case broadcasted vector should
+            have 1 less number of dimensions than the matrix tensor.
+
+    Returns:
+        Two TensorRT ITensors that are broadcasted to the same number of dimensions.
+    """
+    a_shape = tuple(a.shape)
+    b_shape = tuple(b.shape)
+
+    diff = len(a_shape) - len(b_shape) - preset_diff
+    if diff > 0:
+        b = prepend_ones(network, b, f"{b_name}_broadcast", diff)
+    elif diff < 0:
+        a = prepend_ones(network, a, f"{a_name}_broadcast", -diff)
+
+    return a, b
+
+
+def get_axes_for_reduce_op(
+    dim: Union[int, Sequence[int]],
+    has_implicit_batch_dimension: bool = False,
+) -> int:
+    """
+    TensorRT reduce layer relies on the binary representation of axes to
+    determine which dims to reduce. For example, if we want to reduce on
+    dim 1 and 2 then axes should be 6(110).
+
+    Args:
+        dim (Union[int, Sequence[int]]): An integer or a sequence of integers
+            that will be used to generate axes for TensorRT.
+        has_implicit_batch_dimension (bool): Whether the TensorRT network is
+            using implicit batch dimension.
+
+    Returns:
+        An integer which binary form can be used as axes for TensorRT reduce
+        layer.
+    """
+    if isinstance(dim, int):
+        dim = (dim,)
+
+    if has_implicit_batch_dimension:
+        assert 0 not in dim, "Can't reduce over batch dimension when it's implicit."
+
+    axes = 0
+    for d in dim:
+        axes |= 1 << (d - (1 if has_implicit_batch_dimension else 0))
+
+    return axes
+
+
+def has_dynamic_shape(shape: Shape) -> bool:
+    """
+    Determine if the given shape has dynamic dim. i.e. if there're -1 in shape.
+
+    Args:
+        shape (Shape): Shape of a tensor. Essentially is a sequence of integers.
+
+    Returns:
+        A boolean value indicates whether there's dynamic dim in the shape.
+    """
+    count = 0
+    for s in shape:
+        count += 1 if s == -1 else 0
+    return count > 0
```

Comment:
return type was changed when after it's copied from fx.converters.has_dynamic_shape
- return count
+ return count > 0

Reply:
We already have a function `has_dynamic_shapes` here : https://github.com/pytorch/TensorRT/blob/5e7ca649e14742504e318c84464b20498652a598/py/torch_tensorrt/dynamo/conversion/_ConverterRegistry.py#L99
Can you merge these two functionalities in here ? The function can accept either `torch.fx.Node` or `ITensor` in your case and should return boolean accordinly